jdaviz/configs/imviz/helper.py

```diff
@@ -79,6 +84,82 @@ def load_data(self, data, parser_reference=None, **kwargs):
             self.app.load_data(
                 data, parser_reference=parser_reference, **kwargs)
 
+    def load_static_regions(self, regions, data_label, **kwargs):
+        """Load given region(s) into the viewer.
+        Once loaded, the region(s) cannot be modified.
+
+        Parameters
+        ----------
+        regions : dict
+            Dictionary mapping desired region name to one of the following:
+
+            * Astropy ``regions`` object
+            * ``photutils`` apertures (limited support until ``photutils``
+              fully supports ``regions``)
+            * Numpy boolean array (shape must match data)
+
+            Region name that starts with "Subset" is forbidden and reserved
+            for internal use only.
+
+        data_label : str
+            Label to retrieve a specific data set from the viewer instance.
+            Glue Subset object representing this region will be created
+            based on this data set.
+
+        kwargs : dict
+            Extra keywords to be passed into the region's ``to_mask`` method.
+            This is ignored if Numpy array is given.
+
+        """
+        # TODO: Refactor after https://github.com/spacetelescope/jdaviz/pull/644
+        # is merged.
+        # Or do I have to use self.app.get_data_from_viewer for some reason?
+        data = self.app.data_collection[
+            self.app.data_collection.labels.index(data_label)]
+
+        for subset_label, region in regions.items():
+            if subset_label.startswith('Subset'):
+                warnings.warn(f'{subset_label} is not allowed, skipping. '
+                              'Do not use region name that starts with Subset.')
+                continue
+
+            if hasattr(region, 'to_pixel'):
+                # TODO: GWCS not yet supported, see
+                # https://github.com/astropy/photutils/issues/1219
+                # https://github.com/astropy/regions/issues/374
```

Comment:
This part of the code needs updating when new release of `regions` with GWCS support is done, see https://github.com/astropy/regions/issues/374#issuecomment-856946637 .